### DISCUSSION TITLE
Add source code to a different project

### DISCUSSION THREAD

**Discussion by Unskilledcrab** [dotnet/roslyn#56516] [Q&A]
If I had the following two projects in my solution 

- AspNetCore used as an API
- net standard used as a http client library

Neither of the projects have references to each other

And I wanted to generate httpclients based off my controllers in my API project. Would I be able to have a generator run in the API project but have it add files to the http client library?

This way I would always have an up to date client library to be consumed by users.

**Comment by jasonmalinowski** [comment]
No, we don't support that -- practically, each pass of the compiler only sees it's own inputs -- one project's compilation won't see an unrelated project's inputs.

**Reply by Unskilledcrab** [reply]
Is there anyway to inject code into another project during the compilation process?

Maybe writing files to disk and having another project pick up those files for their own compilation somehow?

**Reply by Unskilledcrab** [reply]
Or, as an alternative, completely create and build your own library project from another projects source? So much of this project is boiler plate but I don't want to have references to my data access layer DLL references in my client code but I want to build some of my client libraries from my data access layer code
